client/src/components/SignIn/index.js

```diff
@@ -34,9 +33,9 @@ class SignInFormBase extends Component {
 
     this.props.firebase
       .doSignInWithEmailAndPassword(email, password)
-      .then(() => {
+      .then(authUser => {
         this.setState({ ...INITIAL_STATE });
-        this.props.history.push(ROUTES.DASHBOARD);
+        this.props.history.push(`/dashboard/${authUser.user.uid}`);
```

Comment:
What happens when a signed-in user puts another person's user ID into the URL? Will they see the other person's games?

Reply:
+1 for safety concerns, but could address this later if it is hard.

Reply:
Good point, @tomatocat. TL;DR, we address this with Firestore security rules.

Longer version:

I believe any attempt to secure your data should be focused on the server side instead of the client side. Whatever checks are employed at the frontend, an attacker can simply change the javascript to skip it, or they can directly issue a request to our backend, not using our frontend at all.

Here's a high level picture as I understand it. When a user signs in with Firebase, Firebase client library exchanges the user credential with Firebase server for a piece of information containing the user info (authUser.user.uid etc) and a signature signing the piece of information using Firebase's private key. When Firebase client library issues request to, say, Firestore, it attaches the piece of information (a concrete representation can be a JWT token) with the request. The server (Firestore, or it can well be our own server), validates the piece of information is authentic by checking the signature against Firebase's public key. At the end of this, the server knows the ID of the sign in user issuing the request, securely (i.e. *authentication* is done). 

Then to your concerns, the *authorization* (See https://www.okta.com/identity-101/authentication-vs-authorization/) is performed at the server. Firestore (or our server) should perform a check against the uid of the signed in user and the document it is fetching. e.g. https://firebase.google.com/docs/firestore/security/rules-conditions#authentication

Reply:
Thank you for the detailed explanation @blackhogz! Unfortunately, @tomatocat is right in that if you were to put in another user's ID into the url, you would see their data. The user key IS auto-generated though, so the likelihood of accidentally doing this is low. As for someone intentionally trying to hack into the system, that might be something for the server side like Yueyao said. Currently passwords are not stored in the database though, and it seems like even if you were to access someone else's dashboard, you can't change their password.